src/core/theme/SuomifiTheme/SuomifiTheme.ts

```diff
@@ -0,0 +1,205 @@
+  ColorToken,
+  RawSpacingDesignTokens,
+  RawTypographyDesignTokens,
+  TypographyToken,
+} from 'suomifi-design-tokens';
+
+export type GradientDesignTokens = typeof derivedTokens.gradients;
+export type FocusDesignTokens = typeof derivedTokens.focus;
+export type RadiusDesignTokens = typeof radius;
+export type ShadowDesignTokens = typeof derivedTokens.shadows;
+export type TransitionDesignTokens = typeof transitions;
+export type ZIndexDesignTokens = typeof zindexes;
+export type SuomifiDesignTokens = typeof desingTokens;
+
+export type ColorProp = keyof ColorDesignTokens;
+export type TypographyProp = keyof TypographyDesignTokens;
+export type SpacingProp = keyof SpacingDesignTokens | '0';
+
+/**
+ * SuomifiTheme
+ * General styles and design tokens for suomifi-ui-components
+ */
+export interface SuomifiTheme {
+  /** Color design tokens as [key:string]: CSS string format with color name as key. */
+  colors: ColorDesignTokens;
+  /** Gradient design tokens as [key:string]: CSS string format with key derived from gradient name colors */
+  gradients: GradientDesignTokens;
+  /** Focus design tokens as [key:string]: CSS string format for pseudo and regular styles and disabling mouse focus  */
+  focus: FocusDesignTokens;
+  /** Radius design tokens as [key:string]: CSS string format with radius use case name as key */
+  radius: RadiusDesignTokens;
+  /** Shadow design tokens as [key:string]: CSS string format with shadow type name as key */
+  shadows: ShadowDesignTokens;
+  /** Spacing design tokens as [key:string]: CSS string format with spacing size name as key */
+  spacing: SpacingDesignTokens;
+  /** Transition design tokens as [key:string]: CSS string format with transition type name as key */
+  transitions: TransitionDesignTokens;
+  /** Typography design tokens as [key:string]: CSS string format with text type name as key */
+  typography: TypographyDesignTokens;
+  /**
+   * Colors, typography and spacing design token values in granular format for custom composition of styles
+   * Colors as [key: string]: ColorToken with color name as key.
+   * Typography as [key: string]: TypographyToken with text type as key
+   * Spacing as [key: string]: ValueUnit with spacing size name as key.
+   */
+  values: RawDesignTokens;
+  /** Z-idnex design tokens as [key:string]: CSS string format with z-index type name as key */
+  zindexes: ZIndexDesignTokens;
+}
+
+export interface SuomifiThemeProp {
+  theme: SuomifiTheme;
+}
+
+const desingTokens = {
+  zindexes,
+  transitions,
+  radius,
+  colors: suomifiDesignTokens.colors,
+  spacing: suomifiDesignTokens.spacing,
+  typography: suomifiDesignTokens.typography,
+  values: suomifiDesignTokens.values,
+};
+
+const derivedTokens = {
+  focus: {
+    absoluteFocus: absoluteFocus(desingTokens),
+    boxShadowFocus: boxShadowFocus(desingTokens),
+    noMouseFocus,
+  },
+  shadows: shadows(desingTokens.colors),
+  gradients: gradients(desingTokens.colors),
+};
+
+/** Suomi.fi theme */
```

Comment:
Maybe changing this
```
Suomi.fi theme
```
Also to be like it seems to be in other places

```
SuomifiTheme
```

Reply:
Changed and added short explanation referring to default styles